filter/filter.go

```diff
@@ -478,6 +479,29 @@ func (c *Client) GetDimensionOptionsBatchProcess(ctx context.Context, userAuthTo
 	return eTag, batch.ProcessInConcurrentBatches(batchGetter, batchProcessor, batchSize, maxWorkers)
 }
 
+// DeleteDimensionOptions completely removes the options array from a given dimension
+func (c *Client) DeleteDimensionOptions(ctx context.Context, userAuthToken, serviceAuthToken, collectionID, filterID, name string) (string, error) {
+	uri := fmt.Sprintf("%s/filters/%s/dimensions/%s/options", c.hcCli.URL, filterID, name)
+	clientlog.Do(ctx, "deleting selected dimension options", service, uri)
+
+	resp, err := c.doDeleteWithAuthHeadersAndWithDownloadToken(ctx, userAuthToken, serviceAuthToken, collectionID, uri)
+	if err != nil {
+		return "", errors.Wrap(err, "an error occurred when attempting to delete dimension options")
```

Comment:
I won't say this is a blocker this time but in general I think error messages should be as concise as possible. `failed to` is preferable to `an error occurred when attempting to` especially as these errors generally get stitched together to form long chains, things can quickly get overly convoluted.

Reply:
This has been addressed.